drone-webdriver/src/test/java/org/jboss/arquillian/drone/webdriver/factory/HtmlUnitDriverTest.java

```diff
@@ -0,0 +1,76 @@
+package org.jboss.arquillian.drone.webdriver.factory;
+
+import com.gargoylesoftware.htmlunit.FailingHttpStatusCodeException;
+import com.gargoylesoftware.htmlunit.MockWebConnection;
+import com.gargoylesoftware.htmlunit.WebClient;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.URL;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import org.jboss.arquillian.drone.webdriver.configuration.WebDriverConfiguration;
+import org.jboss.arquillian.drone.webdriver.htmlunit.DroneHtmlUnitDriver;
+import org.jboss.arquillian.drone.webdriver.utils.Validate;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.phantomjs.PhantomJSDriver;
+import org.openqa.selenium.phantomjs.PhantomJSDriverService;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.jboss.arquillian.drone.webdriver.factory.HtmlUnitDriverFactory.webClientOptions;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HtmlUnitDriverTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void shouldOverrideWebClientOptions() throws IOException {
```

Comment:
Interesting test.
What I had in my mind was just creating an instance of the DroneHtmlUnitDriver (as you did) and then get the options and verify the content:
~~~java
WebClientOptions options = htmlUnitDriver.getWebClient().getOptions();
Assertions.assertThat(options.isThrowExceptionOnFailingStatusCode()).isFalse();
~~~ 
So, only verify if the properties have been parsed and set to the WebClient.

Just for clarification - your test if fine - keep it as it is, just add another one for the verification of the options.